Stock.Api/Controllers/ProviderController.cs

```diff
@@ -0,0 +1,128 @@
+        private ProviderService service;
+        private readonly IMapper mapper;
+        public ProviderController(ProviderService service, IMapper mapper)
+        {
+            this.service = service;
+            this.mapper = mapper;
+        }
+        /// <summary>
+        /// Permite recuperar todas las instancias
+        /// </summary>
+        [HttpGet]
+        public ActionResult<IEnumerable<ProviderDTO>> Get()
+        {
+            try
+            {
+                var result = this.service.GetAll();
+                return this.mapper.Map<IEnumerable<ProviderDTO>>(result).ToList();
+            }
+            catch(Exception)
+            {
+                return StatusCode(500);
+            }
+        }
+        /// <summary>
+        /// Permite recuperar una instancia mediante un identificador
+        /// </summary>
+        /// <param name="id">Identificador de la instancia a recuperar</param>
+        /// <returns>Una instancia</returns>
+        [HttpGet("{id}")]
+        public ActionResult<ProviderDTO> Get(string id)
+        {
+            try
+            {
+                var result = this.service.Get(id);
+                return this.mapper.Map<ProviderDTO>(result);
+            }
+            catch (Exception)
+            {
+                return StatusCode(500);
+            }
+        }
+        /// <summary>
+        /// Permite crear un provider
+        /// </summary>
+        /// <param name ="value">Una Instancia</param> 
+        [HttpPost]
+        public ActionResult Post([FromBody] ProviderDTO value){
+            TryValidateModel(value);
+            try{
+                var provider = this.mapper.Map<Provider>(value);
+                this.service.Create(provider);
+                value.Id = provider.Id;
+                return Ok(new {Success = true, Message = "", data = value});
+            }
+            catch{
+                return Ok(new {Success = false, Message = "The name is already in used"});
+            }
+        }
+        /// <summary>
+        /// Permite editar una instancia
+        /// </summary>
+        /// <param name="id">Identificador de la instancia a editar</param>
+        /// <param name="value">Una instancia con los nuevos datos</param>
+        [HttpPut("{id}")]
+        public void Put(string id, [FromBody] ProviderDTO value)
+        {
+            var provider = this.service.Get(id);
+            TryValidateModel(value);
+            this.mapper.Map<ProviderDTO,Provider>(value,provider);
+            this.service.Update(provider);
+        }
+        /// <summary>
+        /// Permite borrar una instancia
+        /// </summary>
+        /// <param name="id">Identificador de la instancia a borrar</param>
+        [HttpDelete("{id}")]
+        public ActionResult Delete(string id)
+        {
+            var provider = this.service.Get(id);
+            this.service.Delete(provider);
+            return Ok(new{ Success = true, Message = "", data = id});
```

Comment:
try and catch para manejar errores  y las respuestas en casos de falla

Reply:
Ok, Lo voy a corregir.